apps/field.go

```diff
@@ -3,39 +3,71 @@ package apps
 type FieldType string
 
 const (
-	FieldTypeText          = FieldType("text")
-	FieldTypeStaticSelect  = FieldType("static_select")
-	FieldTypeDynamicSelect = FieldType("dynamic_select")
-	FieldTypeBool          = FieldType("bool")
-	FieldTypeUser          = FieldType("user")
-	FieldTypeChannel       = FieldType("channel")
+	// Text field. "subtype":"textarea" for multi-line text input (Modal only?).
+	// min_length, max_length - self explanatory. Value (default) is expected to be type string, or nil.
+	FieldTypeText FieldType = "text"
+
+	// Static select field. The options are specified in "options"
+	FieldTypeStaticSelect FieldType = "static_select"
+
+	// Dynamic select's options are fetched by making a call to the Form's Call,
+	// in lookup mode. It responds with the list of options to display.
+	FieldTypeDynamicSelect FieldType = "dynamic_select"
+
+	// Boolean (checkbox) field.
+	FieldTypeBool FieldType = "bool"
+
+	// A mattermost @username.
+	FieldTypeUser FieldType = "user"
+
+	// A mattermost channel reference (~name).
+	FieldTypeChannel FieldType = "channel"
 )
 
 type SelectOption struct {
-	Label    string `json:"label"`
+	// Label is the display name/label for the option's value.
+	Label string `json:"label"`
+
 	Value    string `json:"value"`
 	IconData string `json:"icon_data"`
```

Comment:
It is called iconData, but probably we should rename this to IconPath or IconURL. Nevertheless, not 100% sure if it is implemented at the moment.